drivers/rtc/rtc_rv8263.c

```diff
@@ -424,9 +443,7 @@ static int rv8263c8_alarm_set_time(const struct device *dev, uint16_t id, uint16
 
 	/* Disable the alarm when mask is zero. */
 	if (mask == 0) {
-		return i2c_reg_update_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_CONTROL_2,
-					      RV8263C8_BM_ALARM_INT_ENABLE,
-					      RV8263C8_BM_ALARM_INT_DISABLE);
+		return rv8263c8_disable_alarm(dev);
 	}
```

Comment:
This leaves interrupts incorrectly disabled after disabling and re-enabling the alarm using `rtc_alarm_set_time()` (if the interrupts were previously enabled).

I think you can remove these lines entirely, as setting the AE_x bits is handled correctly by the following code.

Reply:
You are right. If `mask` is set to 0 no alarm will be enabled.

Reply:
But I think I can use it to globally enable or disable the interrupts by setting or clearing the bit in the `CONTROL_2` register. Something like this

```
if (mask == 0) {
i2c_reg_update_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_CONTROL_2,
				    RV8263C8_BM_ALARM_INT_ENABLE, RV8263C8_BM_ALARM_INT_DISABLE);
} else {
i2c_reg_update_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_CONTROL_2,
				    RV8263C8_BM_ALARM_INT_ENABLE, RV8263C8_BM_ALARM_INT_ENABLE);
}
````

What do you think?

Reply:
Sounds good to me :)

Reply:
Ok then I would remove the alarm enabling and disabling completely from the `alarm_set_callback` API and move it to the `arlarm_set_time` API. So the alarm can only be disabled by calling `set_time` and `set_callback` can only remove the callback.

Reply:
Yes! That's certainly feasible, solves the issue above and behaves according to the documentation.